File: extensions/intellij/src/main/resources/continue_tutorial.ts
//                        _________               _____ _____
//                           __  ____/______ _______ __  /____(_)_______ ____  _______
//                           _  /     _  __ \__  __ \_  __/__  / __  __ \_  / / /_  _ \
//                           / /___   / /_/ /_  / / // /_  _  /  _  / / // /_/ / /  __/
//                           \____/   \____/ /_/ /_/ \__/  /_/   /_/ /_/ \__,_/  \___/
//
//                                       Chat, Edit, and Autocomplete tutorial
//

// ————————————————————————————————————————————————     Setup      ————————————————————————————————————————————————-

// First, open the Continue sidebar by pressing [Cmd + L] or clicking the Continue icon.

// See an example at https://docs.continue.dev/getting-started/install

// Follow the instructions in the sidebar to set up models for Chat and Autocomplete.

// —————————————————————————————————————————————————     Chat      —————————————————————————————————————————————————

// Highlight the code below
// Press [Cmd + L] to add to Chat
// Try asking Continue "what sorting algorithm is this?"
function sortingAlgorithm(x: number[]): number[] {
  for (let i = 0; i < x.length; i++) {
    for (let j = 0; j < x.length - 1; j++) {
      if (x[j] > x[j + 1]) {
        let temp = x[j];
        x[j] = x[j + 1];
        x[j + 1] = temp;
      }
    }
  }
  return x;
}

// [Cmd + L] always starts a new chat. Now, try the same thing using [Cmd + Shift + L].
// This will add the code into the current chat

// —————————————————————————————————————————————————     Edit      —————————————————————————————————————————————————

// Highlight the code below
// Press [Cmd + I] to Edit
// Try asking Continue to "make this more readable"
function sortingAlgorithm2(x: number[]): number[] {
  for (let i = 0; i < x.length; i++) {
    for (let j = 0; j < x.length - 1; j++) {
      if (x[j] > x[j + 1]) {
        let temp = x[j];
        x[j] = x[j + 1];
        x[j + 1] = temp;
      }
    }
  }
  return x;
}

// —————————————————————————————————————————————     Autocomplete     ——————————————————————————————————————————————

// Place cursor after `sortingAlgorithm:` below and press [Enter]
// Press [Tab] to accept the Autocomplete suggestion

// Basic assertion for sortingAlgorithm:

// —————————————————————————————————————————————-     Learn More     -——————————————————————————————————————————————

// Visit the Continue Docs at https://docs.continue.dev/getting-started/overview

File: extensions/intellij/src/main/resources/continue_tutorial.java
//                            _________               _____ _____
//                               __  ____/______ _______ __  /____(_)_______ ____  _______
//                               _  /     _  __ \__  __ \_  __/__  / __  __ \_  / / /_  _ \
//                               / /___   / /_/ /_  / / // /_  _  /  _  / / // /_/ / /  __/
//                               \____/   \____/ /_/ /_/ \__/  /_/   /_/ /_/ \__,_/  \___/
//
//                                 Autocomplete, Edit, Chat, and Agent tutorial
//

// —————————————————————————————————————————————     Autocomplete     —————————————————————————————————————————————— //
//                            Autocomplete provides inline code suggestions as you type.

// 1. Place cursor after `sortingAlgorithm:` below and press [Enter]
// 2. Press [Tab] to accept the Autocomplete suggestion

// Basic assertion for sortingAlgorithm:

// —————————————————————————————————————————————————     Edit      ————————————————————————————————————————————————— //
//                   Edit is a convenient way to make quick changes to specific code and files.

// 1. Highlight the code below
// 2. Press [Cmd/Ctrl + I] to Edit
// 3. Try asking Continue to "make this more readable"
public static int[] sortingAlgorithm(int[] x) {
    for (int i = 0; i < x.length; i++) {
        for (int j = 0; j < x.length - 1; j++) {
            if (x[j] > x[j + 1]) {
                int temp = x[j];
                x[j] = x[j + 1];
                x[j + 1] = temp;
            }
        }
    }
    return x;
}

// —————————————————————————————————————————————————     Chat      ————————————————————————————————————————————————— //
//                    Chat makes it easy to ask for help from an LLM without needing to leave the IDE.

// 1. Highlight the code below
// 2. Press [Cmd/Ctrl + L] to add to Chat
// 3. Try asking Continue "what sorting algorithm is this?"
public static int[] sortingAlgorithm2(int[] x) {
    for (int i = 0; i < x.length; i++) {
        for (int j = 0; j < x.length - 1; j++) {
            if (x[j] > x[j + 1]) {
                int temp = x[j];
                x[j] = x[j + 1];
                x[j + 1] = temp;
            }
        }
    }
    return x;
}

// —————————————————————————————————————————————————     Agent      ————————————————————————————————————————————————— //
//         Agent equips the Chat model with the tools needed to handle a wide range of coding tasks,
//         allowing the model to make decisions and save you the work of manually finding context and performing actions.

// 1. Switch from "Chat" to "Agent" mode using the dropdown in the bottom left of the input box
// 2. Try asking Continue "Write unit tests for this code in a new file and run the test"

// ——————————————————      Learn more at https://docs.continue.dev/getting-started/overview      ——————————————————— //